contexts/Aladin.tsx

```diff
@@ -1,127 +1,46 @@
-  defaultHiPSOptions,
-} from "@/fixtures/defaultAladinOptions";
-import { useOnClickOutside } from "@/hooks/listeners";
+import { createContext, RefObject, useContext, useEffect } from "react";
+import { bindAladinEvents } from "@/lib/aladin/helpers";
 
-interface AladinProps {
-  fovRange?: Array<number>;
-  options?: AladinOptions;
-  hipsConfig: {
-    id: string;
-    options: ImageHiPSOptions;
-  };
-}
-
-interface AladinContextValue {
-  aladin: AladinInstance | null;
-  A: Aladin | null;
-  setRef: RefCallback<HTMLElement>;
-  ref: MutableRefObject<HTMLElement | null>;
-  isLoading: boolean;
+export interface AladinContext {
+  isLoading: false;
   hasFocus: boolean;
+  aladin: AladinInstance;
+  A: Aladin;
+  ref: RefObject<HTMLDivElement>;
+}
+export interface AladinContextDefault {
+  isLoading: true;
+  hasFocus: false;
+  aladin?: undefined;
+  A?: undefined;
+  ref?: undefined;
 }
 
-const AladinContext = createContext<AladinContextValue | null>(null);
-
-export const AladinProvider: FunctionComponent<
-  PropsWithChildren<AladinProps>
-> = ({ children, fovRange = [2, 90], hipsConfig, options = {} }) => {
-  const A = useRef<Aladin | null>(null);
-  const aladin = useRef<AladinInstance | null>(null);
-
-  const [hasFocus, setFocus] = useState(false);
-  const [isLoading, setLoading] = useState(true);
-
-  const renderRef = useRef<HTMLElement | null>(null);
-  const setRef: RefCallback<HTMLElement> = useCallback((node: HTMLElement) => {
-    renderRef.current = node;
-  }, []);
-
-  const onFocus = () => {
-    setFocus(true);
-  };
-
-  const onBlur = () => {
-    setFocus(false);
-  };
-
-  useLayoutEffect(() => {
-    let isUnmounted = false;
-
-    (async () => {
-      const global: Aladin = (await import("aladin-lite")).default;
-
-      if (!isUnmounted) {
-        global.init.then(() => {
-          const newInstance: AladinInstance = global.aladin(
-            renderRef?.current || "",
-            Object.assign(defaultOptions, options)
-          );
-          newInstance.setImageSurvey(
-            global.imageHiPS(hipsConfig.id, {
-              ...defaultHiPSOptions,
-              ...hipsConfig.options,
-              successCallback: () => {
-                newInstance.setFoVRange(fovRange[0], fovRange[1]);
-              },
-            })
-          );
-          renderRef?.current?.addEventListener("click", onFocus);
-
-          A.current = global;
-          aladin.current = newInstance;
-          setLoading(false);
-        });
-      }
-    })();
+export const defaultValue: AladinContextDefault = {
+  isLoading: true,
+  hasFocus: false,
+};
 
-    return () => {
-      isUnmounted = true;
-      renderRef?.current?.removeEventListener("click", onFocus);
-      aladin.current = null;
-      A.current = null;
-    };
-  }, []);
+export const AladinContext = createContext<
+  AladinContext | AladinContextDefault
+>(defaultValue);
 
-  useOnClickOutside(renderRef, onBlur);
+export const useAladin: (
```

Comment:
this change is pretty big and narrows the focus of this context down to just the `useAladin` hook, and the provider has been moved to the `Aladin` organism where other concerns like styling are also managed.